parlai/scripts/build_candidates.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""Build the candidate responses for a retrieval model.
+
+Examples
+--------
+
+.. code-block:: shell
+
+  python build_candidates.py -t convai2 --outfile /tmp/cands.txt
+"""
+
+from parlai.core.params import ParlaiParser
+from parlai.agents.repeat_label.repeat_label import RepeatLabelAgent
+from parlai.core.worlds import create_task
+from parlai.core.utils import TimeLogger
+import random
+import tempfile
+
+
+def build_cands(opt):
+    # create repeat label agent and assign it to the specified task
+    agent = RepeatLabelAgent(opt)
+    world = create_task(opt, agent)
+    if opt['outfile'] is None:
+        outfile = tempfile.mkstemp(
+            prefix='{}_{}_'.format(opt['task'], opt['datatype']), suffix='.txt'
+        )[1]
+    else:
+        outfile = opt['outfile']
+
+    if opt['num_examples'] == -1:
+        num_examples = world.num_examples()
+    else:
+        num_examples = opt['num_examples']
+    log_timer = TimeLogger()
+
+    print('[ starting to build candidates from task.. (ex:' +
```

Comment:
this is good; since we axed py35, you can now write:

```python
print(f'[ starting to build candidates from task.. (ex:{num_examples})]')
```

maybe you like this new syntax; it matters not.